### DISCUSSION TITLE
kura.addons.url is 400: Invalid request

### DISCUSSION THREAD

**Discussion by CvEngin** [eclipse-kura/kura#5322] [Q&A] [answered]
When I build, during the process of downloading the relevant configurations from https://raw.githubusercontent.com/eurotech/kura_addons/mvn-repo/, an error occurs and this URL is no longer accessible with the following error message:
Failed to execute goal org.apache.maven.plugins:maven-dependency-plugin:2.9:copy (copy-bundles-for-publishing) on project p2-repo-common: Unable to resolve artifact.: Could not transfer artifact org.eclipse.kura:com.eclipsesource.jaxrs.jersey-min:jar:2.22.2 from/to kura_addons (https://raw.github.com/eurotech/kura_addons/mvn-repo/): GET request of: org/eclipse/kura/com.eclipsesource.jaxrs.jersey-min/2.22.2/com.eclipsesource.jaxrs.jersey-min-2.22.2.jar from kura_addons failed

**Comment by MMaiero** [comment, marked as answer]
Hello,
from time to time this happens and I think this is a limit to downloads set by GitHub.

Maybe give it a try later in the day and see if the process of downloading succeeds.

Best regards,
Matteo

**Reply by CvEngin** [reply]
Hello, thank you very much for your help. That's indeed the reason. But currently, I have encountered another problem and I'm not sure if you have a solution.
Failed to execute goal org.apache.maven.plugins:maven-dependency-plugin:3.0.0:copy (copy-bundles-for-publishing) on project p2-repo-tests-deps: Unable to find/resolve artifact.: Could not transfer artifact org.hamcrest:org.hamcrest:jar:1.1.0.v20090501071000 from/to Metrics (https://mvnrepository.com/artifact/): Access denied to: https://mvnrepository.com/artifact/org/hamcrest/org.hamcrest/1.1.0.v20090501071000/org.hamcrest-1.1.0.v20090501071000.jar , ReasonPhrase:Forbidden.

**Reply by MMaiero** [reply]
Hi, just tried on a clean machine and I cannot recreate your issue.

Maybe give it a try from another machine or maybe your IP is somehow blocked by maven central?

**Reply by CvEngin** [reply]
Hello, yes, I have also tried changing the IP and device, but I still cannot find this jar package https://mvnrepository.com/artifact/org/hamcrest/org.hamcrest/1.1.0.v20090501071000/org.hamcrest-1.1.0.v20090501071000.jar ； And I can't find it in Maven's central repository either. I wonder if this is an issue with the project configuration file

**Reply by MMaiero** [reply]
That should be downloaded from the kura_addons repo. Here a log snippet from my PC:

```
Downloaded from kura_addons: https://raw.github.com/eurotech/kura_addons/mvn-repo/org/hamcrest/org.hamcrest/1.1.0.v20090501071000/org.hamcrest-1.1.0.v20090501071000.jar (8.4 kB at 7.1 kB/s)
```

It is fine that maven central reporter error. Maven should look at all the different repos configured and download where the artefact is available.
Here the full snippet:

```
[INFO] Configured Artifact: org.hamcrest:org.hamcrest:1.1.0.v20090501071000:jar
Downloading from Metrics: https://mvnrepository.com/artifact/org/hamcrest/org.hamcrest/1.1.0.v20090501071000/org.hamcrest-1.1.0.v20090501071000.pom
Downloading from Kura Releases: https://repo.eclipse.org/content/repositories/kura-releases/org/hamcrest/org.hamcrest/1.1.0.v20090501071000/org.hamcrest-1.1.0.v20090501071000.pom
Downloading from Kura Snapshots: https://repo.eclipse.org/content/repositories/kura-snapshots/org/hamcrest/org.hamcrest/1.1.0.v20090501071000/org.hamcrest-1.1.0.v20090501071000.pom
Downloading from kura_addons: https://raw.github.com/eurotech/kura_addons/mvn-repo/org/hamcrest/org.hamcrest/1.1.0.v20090501071000/org.hamcrest-1.1.0.v20090501071000.pom
Downloaded from kura_addons: https://raw.github.com/eurotech/kura_addons/mvn-repo/org/hamcrest/org.hamcrest/1.1.0.v20090501071000/org.hamcrest-1.1.0.v20090501071000.pom (413 B at 361 B/s)
Downloading from Metrics: https://mvnrepository.com/artifact/org/hamcrest/org.hamcrest/1.1.0.v20090501071000/org.hamcrest-1.1.0.v20090501071000.jar
Downloading from Kura Releases: https://repo.eclipse.org/content/repositories/kura-releases/org/hamcrest/org.hamcrest/1.1.0.v20090501071000/org.hamcrest-1.1.0.v20090501071000.jar
Downloading from Kura Snapshots: https://repo.eclipse.org/content/repositories/kura-snapshots/org/hamcrest/org.hamcrest/1.1.0.v20090501071000/org.hamcrest-1.1.0.v20090501071000.jar
Downloading from kura_addons: https://raw.github.com/eurotech/kura_addons/mvn-repo/org/hamcrest/org.hamcrest/1.1.0.v20090501071000/org.hamcrest-1.1.0.v20090501071000.jar
Downloaded from kura_addons: https://raw.github.com/eurotech/kura_addons/mvn-repo/org/hamcrest/org.hamcrest/1.1.0.v20090501071000/org.hamcrest-1.1.0.v20090501071000.jar (8.4 kB at 7.1 kB/s)
[INFO] Configured Artifact: org.hamcrest:org.hamcrest.integration:1.1.0.v201303031500:jar
```

**Reply by CvEngin** [reply]
It's right, thanks!
